File: common/src/main/java/dev/mayaqq/estrogen/client/registry/blockRenderers/cookieJar/CookieJarRenderer.java
package dev.mayaqq.estrogen.client.registry.blockRenderers.cookieJar;

import com.mojang.blaze3d.vertex.PoseStack;
import com.mojang.math.Axis;
import dev.mayaqq.estrogen.registry.blockEntities.CookieJarBlockEntity;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.blockentity.BlockEntityRenderer;
import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider;
import net.minecraft.client.renderer.entity.ItemRenderer;
import net.minecraft.world.item.ItemDisplayContext;
import net.minecraft.world.item.ItemStack;

public class CookieJarRenderer implements BlockEntityRenderer<CookieJarBlockEntity> {
    private final ItemRenderer itemRenderer;

    public CookieJarRenderer(BlockEntityRendererProvider.Context context) {
        itemRenderer = context.getItemRenderer();
    }

    @Override
    public void render(CookieJarBlockEntity be, float partialTick, PoseStack poseStack, MultiBufferSource bufferSource, int light, int overlay) {
        if (be.getLevel() == null) {
            return;
        }

        poseStack.pushPose();
        poseStack.mulPose(Axis.XN.rotationDegrees(90));
        poseStack.translate(0.5, -0.625, 0.032F);

        for (ItemStack jarItem : be.getItems()) {
            if (jarItem.isEmpty()) {
                continue;
            }
            poseStack.translate(-0.025, -0.025, 0.032F);
            itemRenderer.renderStatic(
                    jarItem, ItemDisplayContext.GROUND,
                    light, overlay, poseStack, bufferSource,
                    be.getLevel(), 0
            );
            poseStack.translate(0.05, 0.05, 0.032F);
            itemRenderer.renderStatic(
                    jarItem, ItemDisplayContext.GROUND,
                    light, overlay, poseStack, bufferSource,
                    be.getLevel(), 0
            );
            poseStack.translate(-0.025, -0.025, 0);
        }

        poseStack.popPose();
    }
}

File: src/main/kotlin/dev/mayaqq/estrogen/client/content/blockRenderers/cookieJar/CookieJarRenderer.kt
package dev.mayaqq.estrogen.client.content.blockRenderers.cookieJar

import com.mojang.blaze3d.vertex.PoseStack
import com.mojang.math.Axis
import dev.mayaqq.estrogen.Estrogen
import dev.mayaqq.estrogen.content.blockEntities.CookieJarBlockEntity
import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.MultiBufferSource
import net.minecraft.client.renderer.blockentity.BlockEntityRenderer
import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider
import net.minecraft.world.item.ItemDisplayContext

class CookieJarRenderer(val ctx: BlockEntityRendererProvider.Context) : BlockEntityRenderer<CookieJarBlockEntity> {

    override fun render(
        be: CookieJarBlockEntity,
        partialTick: Float,
        poseStack: PoseStack,
        bufferSource: MultiBufferSource,
        light: Int,
        overlay: Int
    ) {
        // IDEA is wrong here, block entity level can be null
        if (be.level == null) return

        val itemRenderer = Minecraft.getInstance().itemRenderer

        poseStack.pushPose()
        poseStack.mulPose(Axis.XN.rotationDegrees(90f))
        poseStack.translate(0.5, -0.625, 0.032)

        Estrogen.info(be.count.toString())
        for (jarItem in be.items) {
            if (jarItem.isEmpty) {
                continue
            }
            poseStack.translate(-0.025, -0.025, 0.032)
            itemRenderer.renderStatic(
                jarItem, ItemDisplayContext.GROUND,
                light, overlay, poseStack, bufferSource,
                be.level, 0
            )
            poseStack.translate(0.05, 0.05, 0.032)
            itemRenderer.renderStatic(
                jarItem, ItemDisplayContext.GROUND,
                light, overlay, poseStack, bufferSource,
                be.level, 0
            )
            poseStack.translate(-0.025, -0.025, 0.0)
        }

        poseStack.popPose()
    }
}